securedrop_client/gui/actions.py

```diff
@@ -250,23 +251,55 @@ def __init__(
         parent: QMenu,
         controller: Controller,
         source: Source,
+        app_state: Optional[state.State] = None,
     ) -> None:
         """
-        Allows export of a conversation transcript.
+        Allows export of a conversation transcript and all is files. Will download any file
+        that wasn't already downloaded.
         """
-        text = _("Export Conversation")
+        text = _("Export All")
 
         super().__init__(text, parent)
 
         self.controller = controller
         self._source = source
+        self._state = app_state
 
         self._export_device = ConversationExportDevice(controller)
 
         self.triggered.connect(self._on_triggered)
 
     @pyqtSlot()
     def _on_triggered(self) -> None:
+        """
+        (Re-)generates the conversation transcript and opens a confirmation dialog to export it
+        alongside all the (attached) files that are downloaded, in the manner
+        of the existing ExportFileDialog.
+        """
+        if self.controller.api is None:
```

Comment:
The effect of this is that Export All can only run while the user is online and signed in.  Do you think that's a hard requirement? (I would assume we're checking if they've downloaded all the files that appear in their existing message history to flash the modal.)

Reply:
Fixed!